File: chatagent_app/llm.py
import os
from dotenv import load_dotenv

load_dotenv()

provider = os.getenv("LLM_PROVIDER", "groq").lower()

if provider == "groq":
    from langchain_groq import ChatGroq

    api_key = os.getenv("GROQ_API_KEY")
    if not api_key:
        raise ValueError("GROQ_API_KEY not found in environment variables.")

    llm = ChatGroq(
        groq_api_key=api_key,
        model_name=os.getenv("GROQ_MODEL", "openai/gpt-oss-120b"),
        temperature=0
    )

# elif provider == "openai":
#     from langchain_openai import ChatOpenAI

#     api_key = os.getenv("OPENAI_API_KEY")
#     if not api_key:
#         raise ValueError("OPENAI_API_KEY not found in environment variables.")

#     llm = ChatOpenAI(
#         openai_api_key=api_key,
#         model_name=os.getenv("OPENAI_MODEL", "gpt-4o"),
#         temperature=0
#     )

# else:
#     raise ValueError(f"Unsupported LLM_PROVIDER: {provider}")

File: backend/lib/llm.js
import { ChatGroq } from '@langchain/groq';

const apiKey = process.env.GROQ_API_KEY;
const modelName = process.env.GROQ_MODEL || 'llama-3.3-70b-versatile';

export const llm = apiKey ? new ChatGroq({
    apiKey: apiKey,
    modelName: modelName,
    temperature: 0,
}) : null;

if (!llm) {
    console.warn('⚠️  GROQ_API_KEY is missing! AI agent features will not work.');
}
